src/UIHandler.h

```diff
@@ -0,0 +1,131 @@
+#pragma once
+#ifndef UIHANDLER_H
+#define UIHANDLER_H
+
+#include "debug.h"
```

Comment:
Why do you need that header in this *header* (you may need it in the cpp file, but not in the h file). Similar: you use `std::vector`, but you don't include `<vector>` anywhere. Don't rely on the other header to include what you use, include it yourself.